plugins/signals-backend/src/service/SignalManager.ts

```diff
@@ -64,6 +68,46 @@ export class SignalManager {
       onEvent: (params: EventParams) =>
         this.onEventBrokerEvent(params.eventPayload as SignalPayload),
     });
+
+    if (
+      options.config.getOptionalBoolean('signals.server.pingEnabled') ??
+      true
+    ) {
+      const pingIntervalConfig = options.config.getOptionalConfig(
+        'signals.server.pingInterval',
+      );
+      this.pingInterval = setInterval(
+        () => this.ping(),
+        pingIntervalConfig
+          ? durationToMilliseconds(readDurationFromConfig(pingIntervalConfig))
+          : 30000,
+      );
+    }
+  }
+
+  private ping() {
+    this.connections.forEach(conn => {
+      if (!conn.isAlive) {
+        this.logger.info(`Connection ${conn.id} is not alive, closing`);
+        conn.ws.terminate();
+        return;
+      }
+
+      conn.isAlive = false;
+      this.logger.debug(`Pinging connection ${conn.id}`);
+      conn.ws.ping();
```

Comment:
We'd really want to wait for the `pong` as well with some form of timeout I'm thinking? To make sure we're not just sending pings into the void

Reply:
And, the timeout to wait for the pong is basically the `setInterval` timeout as when it's next triggered -> if the `isAlive` is false, the connection is terminated. So basically the client has 30 seconds time to respond to ping or it will be terminated.

Reply:
Ah alright, perhaps a bit long to wait for the pong, but very simple 👍 Let's :shipit: